src/network/postgres_protocol_handler.cpp

```diff
@@ -258,26 +258,24 @@ ProcessResult PostgresProtocolHandler::ExecQueryMessage(InputPacket *pkt, const
     std::vector<StatementResult> result;
     std::vector<FieldInfo> tuple_descriptor;
     std::string error_message;
-    int rows_affected = 0;
 
     std::string query_type_string_;
     Statement::ParseQueryTypeString(query, query_type_string_);
 
     QueryType query_type;
     Statement::MapToQueryType(query_type_string_, query_type);
 
-    query_ = query;
-    query_type_ = query_type;
+    traffic_cop_->query_ = query;
 
     switch (query_type) {
       case QueryType::QUERY_PREPARE:
       {
         std::string statement_name;
         std::vector<std::string> tokens;
-        boost::split(tokens, query_, boost::is_any_of("(), "));
+        boost::split(tokens, traffic_cop_->query_, boost::is_any_of("(), "));
```

Comment:
We might want to have query local variable that deal with all of this and then set the query_ in traffic_cop by setter

Reply:
Actually, we can get it from statement_, but first, we should solve hard code part "std::size_t pos = boost::to_upper_copy(query_).find("AS");
        std::string statement_query = query_.substr(pos + 3);
        boost::trim(statement_query);" in postgres_protocal_handler

Reply:
Then add a TODO comment on the query_ illustrating this. We could fix it afterward

Reply:
#874 is dealing with hardcoded problem.